### ISSUE TITLE
List.toArray should propagate @MinLen annotation from receiver to result

### ISSUE THREAD

**Issue by mernst** [kelloggm/checker-framework#54]
The result of `List.toArray` is, conservatively, `@MinLen(0)`.  However, it should be the same as the `@MinLen` annotation on its receiver.  This should be able to be handled by adding a `@PolyMinLen` annotation in the JDK.

Test case:
```java
  public String /*@MinLen(1)*/ [] m(/*@MinLen(1)*/ ArrayList<String> compiler) {
    return compiler.toArray(new String[0]);
  }
```

**Comment by mernst** [comment]
Test case added to repository.  Issue closed, because List support is not a priority right now.

**Comment by mernst** [comment]
I'm reopening this because the JDK annoattions need to be removed or fixed.
See this code in plume-lib's `FileCompiler.java`:

```java
  public FileCompiler(/*@MinLen(1)*/ ArrayList<String> compiler, /*@Positive*/ long timeLimit) {
    this(compiler.toArray(new String[0]), timeLimit); // index TODO: issue #54
  }
```

**Comment by kelloggm** [comment]
Blocked on #74

**Comment by kelloggm** [comment]
An index JDK without list support was added in https://github.com/typetools/checker-framework/pull/1034, so this should be fixed.
